### PR TITLE
T694: Fix unit test on 32-bit systems with 64-bit file sizes

### PR DESCRIPTION
**Pull request by kgaillot**
... along with a major cleanup of configure.ac

### CODE REVIEW

File: lib/common/mock.c [hunk 1]
```diff
@@ -285,6 +287,26 @@ __wrap_fopen(const char *pathname, const char *mode)
     }
 }
 
+#ifdef HAVE_FOPEN64
```
Comment (nrwahl2) [inline lib/common/mock.c:290]: Alternatively, `__wrap_fopen()` could go in an `#else` CPP block, with this block guarded by `#if defined(HAVE_FOPEN64) && defined(_FILE_OFFSET_BITS) && (_FILE_OFFSET_BITS == 64) && (SIZEOF_LONG < 8)`... which is more verbose, granted.

I haven't looked into the details of the situation, but it seems like we don't actually need `__wrap_fopen()` if we're using `__wrap_fopen64()`. The better readability of the CPP directives might be worth it though.

File: configure.ac [hunk 2]
```diff
@@ -1115,6 +1112,7 @@ dnl ========================================================================
 
 REQUIRE_FUNC([getopt])
 REQUIRE_FUNC([setenv])
+REQUIRE_FUNC([uuid_unparse])
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply configure.ac:1149]: Whoops!

REQUIRE_FUNC() is more accurately named than AC_MSG_ERROR(), which exits configure with an error status :)

File: configure.ac [hunk 3]
```diff
@@ -20,19 +20,6 @@ m4_ifdef([AC_CONFIG_MACRO_DIRS],
          [AC_CONFIG_MACRO_DIRS([m4])],
          [AC_CONFIG_MACRO_DIR([m4])])
 
-AC_DEFUN([AC_DATAROOTDIR_CHECKED])
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply configure.ac:23]: I revisited this, and we actually do need to define the macro. However we aren't processing datarootdir so I'm adding a commit to clean it up better.

File: configure.ac [hunk 4]
```diff
@@ -72,15 +59,6 @@ AS_IF([test x"${PKG_CONFIG}" != x""], [],
 PKG_INSTALLDIR
 PKG_NOARCH_INSTALLDIR
 
-dnl Example 2.4. Silent Custom Rule to Generate a File
-dnl %-bar.pc: %.pc
-dnl	$(AM_V_GEN)$(LN_S) $(notdir $^) $@
-
-CC_IN_CONFIGURE=yes
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply configure.ac:79]: It was copy-and-paste from another project

File: configure.ac [hunk 5]
```diff
@@ -24,24 +27,22 @@ m4_include([m4/version.m4])
 AC_INIT([pacemaker], VERSION_NUMBER, [users@clusterlabs.org], [pacemaker],
         PCMK_URL)
 
-PCMK_FEATURES=""
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply configure.ac:40]: True, I just felt it didn't belong in autotools bootstrapping. I think it's reasonable here.

File: configure.ac [hunk 6]
```diff
@@ -656,6 +654,7 @@ expand_path_option bindir
 expand_path_option sbindir
 expand_path_option libexecdir
 expand_path_option datadir
+expand_path_option datarootdir
```
Comment (nrwahl2) [inline configure.ac:744]: > directories such as datadir are defined in terms of [datarootdir]

Does that mean `datarootdir` should get expanded first? Presumably it makes no actual difference for our purposes.

(You'll notice a pattern... I'd rather ask potentially dumb questions than go learn Autoconf.)

**Review by nrwahl2 — APPROVED** [review]

File: configure.ac (hunk 6 above)
Comment (kgaillot) [reply configure.ac:744]: Ah yes, it should

**Comment by kgaillot** [comment]
Updated per review (and to fix a whitespace issue)

**Review by nrwahl2 — APPROVED** [review]

**Pull request merged**